add_mex is not working on my mac

yet another one.  this happens on a completely clean build on master on my home laptop:

```
[ 21%] Linking CXX shared module /Users/russt/drake-distro/drake/util/barycentricInterpolation.mexmaci64
Undefined symbols for architecture x86_64:
  "_mexErrMsgIdAndTxt", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mexPrintf", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxCreateDoubleMatrix_730", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxCreateNumericMatrix_730", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetCell_730", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetData", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetM", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetN", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetNumberOfElements", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
  "_mxGetPr", referenced from:
      _mexFunction in barycentricInterpolation.cpp.o
```

because (with VERBOSE=1) it's not actually linking in the matlab libs
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -Werror=all -Werror=ignored-qualifiers -DGTEST_DONT_DEFINE_FAIL=1 -DGTEST_DONT_DEFINE_SUCCEED=1 -DGTEST_DONT_DEFINE_TEST=1 -bundle -Wl,-headerpad_max_install_names  -o /Users/russt/drake-distro/drake/util/barycentricInterpolation.mexmaci64 CMakeFiles/barycentricInterpolation.dir/barycentricInterpolation.cpp.o  -L/Users/russt/drake-distro/build/drake/lib  -L/Users/russt/drake-distro/build/install/lib
```

if it helps, the only occurence of the mex libraries I could find were:
```
drake008% more CMakeCache.txt| grep -i libmex
Matlab_MEX_LIBRARY:FILEPATH=/Applications/MATLAB_R2015b.app/bin/maci64/libmex.dylib
FIND_PACKAGE_MESSAGE_DETAILS_Matlab:INTERNAL=[/Applications/MATLAB_R2015b.app/extern/include][/Applications/MATLAB_R2015b.app/bin/maci64/libmex.dylib][mexmaci64][/Applications/MATLAB_R2015b.app][cfound components:  MAIN_PROGRAM ][v8.6()]
```

oddly enough, i do see
```
drake008% grep -R MEX_ * | grep bary
util/CMakeFiles/barycentricInterpolation.dir/flags.make:CXX_FLAGS =  -Werror=all -Werror=ignored-qualifiers -DGTEST_DONT_DEFINE_FAIL=1 -DGTEST_DONT_DEFINE_SUCCEED=1 -DGTEST_DONT_DEFINE_TEST=1 -fPIC   -DMATLAB_MEX_FILE -I"/Applications/MATLAB_R2015b.app/extern/include" -I"/Applications/MATLAB_R2015b.app/simulink/include" -fno-common -arch x86_64 -mmacosx-version-min=10.11 -fexceptions -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -DMATLAB_MEX_FILE -DMATLAB_MEX_FILE -DMATLAB_MEX_FILE  -std=gnu++14
```

how am i supposed to build just drake with the new out-of-source build scripts?
if i run build in drake-distro/build/drake, then it appears to do the right thing, but fails with the error above.  if i cd into drake-distro/drake and run make, then i run into different issues:
```
drake008% make
Configuring with CMAKE_FLAGS: -DCMAKE_BUILD_TYPE=Release
-- CMAKE_INSTALL_PREFIX=/Users/russt/drake-distro/build
-- Found matlab
CMake Warning at CMakeLists.txt:157 (message):
  MATLAB JVM version (1.7) does not match installed Java SDK version (1.8)


CMake Error at /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/FindGTest.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:224 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/russt/drake-distro/drake/pod-build/CMakeFiles/CMakeOutput.log".
make: *** [configure] Error 1
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ endif()
 # Set Default build type to Release if none specified
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING
-    "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel."
+    "The type of build. Options are: Debug Release RelWithDebInfo MinSizeRel."
     FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,6 +13,13 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERS
   message(FATAL_ERROR "MSVC version must be at least 14")
 endif()
 
+# Set Default build type to Release if none specified
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+    "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel."
+    FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
 # PODs out-of-source build logic
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   if(POD_BUILD)

--- a/drake/doc/from_source.rst
+++ b/drake/doc/from_source.rst
@@ -120,12 +120,15 @@ To build with ``Make``, execute::
     cd drake-distro
     mkdir build
     cd build
-    cmake ../ -DCMAKE_BUILD_TYPE:STRING=Debug
+    cmake ../
     make
 
 **Do NOT use sudo.** Just ``make`` is sufficient, and will prevent problems
 later. Feel free to use ``make -j`` if your platform supports it. Note that the
-above ``cmake`` command uses build type "Debug". Alternative build modes include
+above ``cmake`` command does not specify a build type, so drake will be built
+with ``Release`` by default. If you wish to build with a different build type,
+change the cmake command to ``cmake ../ -DCMAKE_BUILD_TYPE:STRING=Debug``
+to build with build type "Debug". Alternative build modes include
 "RelWithDebInfo" and "Release". They differ in terms of the amount of debug
 symbols included in the resulting binaries and how efficiently the code
 executes.
@@ -165,7 +168,7 @@ Ninja build.
     cd drake-distro
     mkdir build
     cd build
-    cmake ../ -G Ninja -DCMAKE_BUILD_TYPE:STRING=Debug
+    cmake ../ -G Ninja
     ninja
 
 Ninja can rebuild Drake from within ``drake-distro/build/drake/`` without

